doc/user_guide/signal/signal_basics.rst

```diff
@@ -133,111 +129,155 @@ example we create a Signal2D instance from a 2D numpy array:
+available when loading the ``hyperspy.api`` into the namespace. A `list of packages
+that extend HyperSpy <https://github.com/hyperspy/hyperspy-extensions-list>`_
+is curated in a dedicated repository.
+
+The metadata attribute ``signal_type`` describes the nature of the signal. It can
+be any string, normally the acronym associated with a particular signal. To print
+all :py:class:`~.api.signals.BaseSignal` subclasses available in your system call
+the function :py:func:`~.api.print_known_signal_types` as in the following
+example:
 
 .. code-block:: python
 
     >>> hs.print_known_signal_types()
     +--------------------+---------------------+--------------------+----------+
     |    signal_type     |       aliases       |     class name     | package  |
     +--------------------+---------------------+--------------------+----------+
-    | DielectricFunction | dielectric function | DielectricFunction | hyperspy |
-    |      EDS_SEM       |                     |   EDSSEMSpectrum   | hyperspy |
-    |      EDS_TEM       |                     |   EDSTEMSpectrum   | hyperspy |
-    |        EELS        |       TEM EELS      |    EELSSpectrum    | hyperspy |
-    |      hologram      |                     |   HologramImage    | hyperspy |
-    |      MySignal      |                     |      MySignal      | hspy_ext |
+    | DielectricFunction | dielectric function | DielectricFunction | elexspy  |
+    |      EDS_SEM       |                     |   EDSSEMSpectrum   | elexspy  |
+    |      EDS_TEM       |                     |   EDSTEMSpectrum   | elexspy  |
+    |        EELS        |       TEM EELS      |    EELSSpectrum    | elexspy  |
+    |      hologram      |                     |   HologramImage    | holospy  |
     +--------------------+---------------------+--------------------+----------+
 
-.. warning::
-    From version 2.0 HyperSpy will no longer ship
-    :py:class:`~.api.signals.BaseSignal` subclasses that are specific to a
-    particular type of data (i.e. with non-empty ``signal_type``). All those
-    signals currently distributed with HyperSpy will be moved to new
-    packages.
+When :ref:`loading data <loading_files>`, the ``signal_type`` will be
+set automatically by the file reader, as defined in ``rosettasciio``. If the
+extension providing the corresponding signal subclass is installed,
+:py:func:`~.api.load` will return the subclass from the hyperspy extension,
+otherwise a warning will be raised to explain that
+no registered signal class can be assigned to the given ``signal_type``.
+
+Functions/methods provided by the extensions that directly operate on the data
+types (thus being object oriented) defined in a registered subclass will be available
+after loading the ``hyperspy.api``. However, the packages might also provide some
+additional utilities, which require them to be separately loaded into the python
+namespace. Check the user guides of the `HyperSpy extensions
```

Comment:
I think that we should focus on what returns `hs.load` rather than importing `hyperspy`. Returning objects provided by other libraries is very common (for example, function that return numpy array), what is special here is that `hs.load` return subclasses of `BaseSignal` implemented in different libraries.
What's about the following?
```suggestion
Since the :py:func:`~.api.load` can return domain specific signal objects (e.g. ``EDSSEMSpectrum`` from ``EleXSpy``) provided by extensions, the corresponding functionalities (so-called `method` of `object` in object oriented programming, e.g. ``EDSSEMSpectrum.get_lines_intensity``) implemented in signal classes of the extension can be accessed readily.
To use additional functionalities implemented in extension but not as method of the signal class, the extensions will need to be imported explicitly (e.g. ``import EleXSpy``). Check the user guides of the `HyperSpy extensions
```